scripts/db/src/get-user.ts

```diff
@@ -0,0 +1,74 @@
+import { Client, ConnectionConfig } from 'pg'
+
+// TODO: This is pretty copy-pasta-y from login. We should fix this by building
+// good code-sharing functionality into this repo. But notice that we can't just
+// naively import and run a function. So we'll need to write code that takes
+// function definitions and inserts them into the top of the function. It is my
+// (JP's) opinion that we should wait until there's one more instance of code
+// re-use before making that refactor.
+
+async function getByEmail(email: string, callback: DbScriptCallback) {
+  // Auth0 comment:
+  // This script should retrieve a user profile from your existing database,
+  // without authenticating the user.
+  // It is used to check if a user exists before executing flows that do not
+  // require authentication (signup and password reset).
+  //
+  // There are three ways this script can finish:
+  // 1. A user was successfully found. The profile should be in the following
+  // format: https://auth0.com/docs/users/normalized/auth0/normalized-user-profile-schema.
+  //     callback(null, profile);
+  // 2. A user was not found
+  //     callback(null);
+  // 3. Something went wrong while trying to reach your database:
+  //     callback(new Error("my error message"));
+  try {
+    /** Get required dependency */
+    const { Client } = require('pg@7.17.1')
+
+    /** Declare connection info */
+    const connectionInfo: ConnectionConfig = {
+      user: configuration.POSTGRES_USERNAME,
+      password: configuration.POSTGRES_PASSWORD,
+      host: configuration.POSTGRES_HOST,
+      database: configuration.POSTGRES_DATABASE,
+      port: configuration.POSTGRES_PORT
+        ? parseInt(configuration.POSTGRES_PORT)
+        : 5432,
+      ssl: TEMPLATE_DATA.pgShouldSsl,
+    }
+
+    /** Construct a postgres client and connect to the server */
+    const client: Client = new Client(connectionInfo)
+    await client.connect()
+
+    /** Get the person based on their email, joining on the password table */
+    const query = `
```

Comment:
I think this isn't the right query – you should check what's running in production, but this script doesn't need to check passwords.

I'm pretty sure we just look at `people.person`, but 5% confident that we might actually be checking `people.email_address`, or otherwise doing something to mitigate weirdness if people sign up using a secondary email.

Reply:
Good spotting. I copied it from the production login script, which is joining on the password field. I claim we probably want to change both of them then, away from what's happening on production. Auth0 says

> You may see unexpected behavior if you return differing user profiles in the login and get_user scripts.

I'd rather stay within the API that they expect us to provide. Now, maybe we should keep this PR as a pure refactor, and have a separate PR to achieve the behavior we want? We can discuss tomorrow morning if you want.

For reference, here's the production query:

```
			SELECT 
				id, email, first_name, last_name
			FROM people.person
			where person.email = $1
```

Which could theoretically return something different than the login script, but in practice shouldn't.

I'll update it to that for now, to keep this a pure refactor.